tokens/usecase/tokens_usecase_test.go

```diff
@@ -539,5 +542,418 @@ func (s *TokensUseCaseTestSuite) TestGetMinPoolLiquidityCap() {
+		},
+		{
+			name:  "Invalid scaling factor type",
+			denom: "invalidTypeScalingFactorDenom",
+			denomMetadataMap: map[string]any{
+				"invalidTypeScalingFactorDenom": domain.Token{Precision: 6},
+			},
+			scalingFactorMap: map[int]any{
+				6: "1000000",
+			},
+			expectedResult: osmomath.Dec{},
+			expectedError:  fmt.Errorf("scaling factor for precision (6) and denom (invalidTypeScalingFactorDenom) not found"),
+		},
+	}
+
+	for _, tt := range testcases {
+		s.Run(tt.name, func() {
+			usecase := tokensusecase.NewTokensUsecase(nil, 0, nil)
+			for k, v := range tt.denomMetadataMap {
+				usecase.SetTokenMetadataByChainDenom(k, v)
+			}
+			for k, v := range tt.scalingFactorMap {
+				usecase.SetPrecisionScalingFactorMap(k, v)
+			}
+
+			result, err := usecase.GetChainScalingFactorByDenomMut(tt.denom)
+			if tt.expectedError != nil {
+				s.Require().EqualError(err, tt.expectedError.Error())
+			} else {
+				s.Require().NoError(err)
+			}
+			s.Require().Equal(tt.expectedResult, result)
+		})
+	}
+}
+
+// Tests the GetCoingeckoIdByChainDenom function.
+func (s *TokensUseCaseTestSuite) TestGetCoingeckoIdByChainDenom() {
+	testcases := []struct {
+		name            string
+		chainDenom      string
+		coingeckoIdsMap map[string]any
+		expectedResult  string
+		expectedError   error
+	}{
+		{
+			name:       "Valid chain denom",
+			chainDenom: "validDenom",
+			coingeckoIdsMap: map[string]any{
+				"validDenom": "coingecko-valid",
+			},
+			expectedResult: "coingecko-valid",
+			expectedError:  nil,
+		},
+		{
+			name:       "Chain denom not found",
+			chainDenom: "invalidDenom",
+			coingeckoIdsMap: map[string]any{
+				"validDenom": "coingecko-valid",
+			},
+			expectedResult: "",
+			expectedError:  fmt.Errorf("chain denom not found in chain registry"),
+		},
+		{
+			name:       "Invalid type - not a string",
+			chainDenom: "invalidType",
+			coingeckoIdsMap: map[string]any{
+				"invalidType": 123,
+			},
+			expectedResult: "",
+			expectedError:  fmt.Errorf("coingecko id for chain denom (invalidType) is not of type string"),
+		},
+	}
+
+	for _, tt := range testcases {
+		s.Run(tt.name, func() {
+			usecase := tokensusecase.NewTokensUsecase(nil, 0, nil)
+			for k, v := range tt.coingeckoIdsMap {
+				usecase.SetCoingeckoIDs(k, v)
+			}
+
+			result, err := usecase.GetCoingeckoIdByChainDenom(tt.chainDenom)
+			if tt.expectedError != nil {
+				s.Require().EqualError(err, tt.expectedError.Error())
+			} else {
+				s.Require().NoError(err)
+			}
+			s.Require().Equal(tt.expectedResult, result)
+		})
+	}
+}
+
+// MockTokenLoader is a mock implementation of TokenLoader.
+type MockTokenLoader struct {
+	loadTokens tokensusecase.LoadTokensFunc
+	callCount  int
+	err        error
+}
+
+func (m *MockTokenLoader) FetchAndUpdateTokens() error {
+	m.callCount++
+	return m.err
+}
+
+func (m *MockTokenLoader) CallCount() int {
+	return m.callCount
+}
```

Comment:
Does it make sense moving this into the `domain/mocks` package?

Reply:
Moved to `domain/mocks`. :)